src/core/queryInstance.ts

```diff
@@ -0,0 +1,104 @@
+import { uid, isServer, isDocumentVisible, Console } from './utils'
+import { Query, QueryState } from './query'
+import { BaseQueryConfig } from './types'
+
+// TYPES
+
+type OnStateUpdateFunction<TResult, TError> = (
+  state: QueryState<TResult, TError>
+) => void
+
+// CLASS
+
+export class QueryInstance<TResult, TError> {
+  id: number
+  onStateUpdate: OnStateUpdateFunction<TResult, TError>
+  config: BaseQueryConfig<TResult, TError>
+
+  private query: Query<TResult, TError>
+  private refetchIntervalId?: number
+
+  constructor(
+    query: Query<TResult, TError>,
+    onStateUpdate: OnStateUpdateFunction<TResult, TError>
+  ) {
+    this.id = uid()
+    this.onStateUpdate = onStateUpdate
+    this.query = query
+    this.config = {}
+  }
+
+  clearInterval(): void {
+    clearInterval(this.refetchIntervalId)
+    delete this.refetchIntervalId
```

Comment:
Do you think instead of using `delete`, we should use the type of `refetchIntervalId` as `number | null` and just set it as `null` on init and clear?

Reply:
I don't have a strong opinion on `delete` vs `null` but this could be a separate PR as it is not really related to TS

Reply:
Removed `delete`, they are now set to `undefined`